### PR TITLE
Extensibility: Apply filters without function wrappers

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Raised by @aduth when working on converting all filters for the components to work as HOCs (https://github.com/WordPress/gutenberg/pull/3827#discussion_r156099186):

> Minor: Do we need to export these as functions anymore, or can they just call to `addFilter` when imported?

This PR explores if we can avoid exporting functions that enable filters for blocks. Instead we apply them directly in the same file when corresponding callbacks are defined. 

## How Has This Been Tested?
Unit tested.

## Types of changes
Refactoring.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: blocks/hooks/test/anchor.js [hunk 1]
```diff
@@ -6,41 +6,31 @@ import { noop } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { applyFilters, removeAllFilters } from '@wordpress/hooks';
```
Comment (aduth) [inline blocks/hooks/test/anchor.js]: So the downside to this approach is that we can't clean up lingering handlers after tests?

Comment (gziolo) [reply blocks/hooks/test/anchor.js]: Test files are isolated, so they should be cleaned up anyways. I hope 😄

Comment (aduth) [reply blocks/hooks/test/anchor.js]: >Test files are isolated, so they should be cleaned up anyways. I hope 😄

Hmm, the require cache is cleared between each? Would be (good) news to me if true.

Comment (gziolo) [reply blocks/hooks/test/anchor.js]: I think it isn’t the case only for mocks imported in Jest setup file.

You can enable reseting modules per test using [resetModules](https://facebook.github.io/jest/docs/en/configuration.html#resetmodules-boolean) config option or call it programmatically with the method called the same.

**Comment by gziolo** [comment]
@aduth should I proceed with this PR? Let me know if you like it more? I'm convinced :)

**Review by aduth — APPROVED** [review]

**Pull request merged**